clean-code-challanges/src/main/java/PigLatinTranslator.java

```diff
@@ -18,6 +18,60 @@
 public class PigLatinTranslator {
 
   public String translate(String englishPhrase) {
-    return null;
+
+    String translation = "";
+    String [] englishWords = englishPhrase.toLowerCase().split(" ");
+
+    for (String wordToTranslate : englishWords){
+      translation += translateWord(wordToTranslate);
+      translation += " ";
+    }
+    return translation.trim();
+  }
+
+  private String translateWord (String word){
+    String translatedWord = "";
+
+    if(startsWithVowel(word) || startsWithVowelEdgeCase(word)){
+      translatedWord = word + "ay";
+    } else if (consonantFollowedByQU(word) || startsWithThreeConsonants(word)){
+      translatedWord = word.substring(3) + word.substring(0, 3) + "ay";
+    } else if (startsWithTwoConsonants(word) || word.startsWith("qu")){
+      translatedWord = word.substring(2) + word.substring(0,2) + "ay";
+    } else {
+      translatedWord = word.substring(1) + word.charAt(0) + "ay";
+    }
+    return translatedWord;
+  }
+
+  private boolean startsWithVowel(String word){
+    String vowels = "aeiou";
+    char letter = word.charAt(0);
+    return (vowels.indexOf(letter) != -1);
+  }
+
+  private boolean startsWithVowelEdgeCase(String word){
+    return (word.startsWith("xr") || word.startsWith("yt"));
+  }
+
+  private boolean consonantFollowedByQU(String word){
+    String quCheck = word.substring(1, 3);
+    return quCheck.equals("qu");
+  }
+
+  private boolean startsWithThreeConsonants(String word){
+    String vowels = "aeiou";
+    char letterOne = word.charAt(0);
+    char letterTwo = word.charAt(1);
+    char letterThree = word.charAt(2);
+    return (vowels.indexOf(letterOne) == -1 && vowels.indexOf(letterTwo) == -1 && vowels.indexOf(letterThree) == -1);
```

Comment:
vowel.contains(letterOne/Two/Three) würde die Sache leserlicher machen (siehe unten, macht eigendlich was du hier ausschreibst). Aber sonst sehr sehr cleane und verständliche sache.
Nebensache: Code Wiederholung in startsWithTwoConsonants etwas schade. 
   
 public boolean contains(CharSequence s) {  
          return indexOf(s.toString()) > -1;  
      }

Reply:
Neue Methode implementiert um die ersten Konsonanten vor einem Vokal zu finden